pkg/datastore.go

```diff
@@ -117,6 +126,27 @@ func (s *DefaultDatastore) getHeaders(ctx context.Context, key ds.Key) ([]Header
 	return s.unsmarshalHeaders(data)
 }
 
+func (s *DefaultDatastore) filterHeaders(headers []HeaderAndTrace, query Query) []HeaderAndTrace {
+	var filtered []HeaderAndTrace
+	for _, header := range headers {
```

Comment:
This would be a good place to do some [zero-allocation filtering](https://zchee.github.io/golang-wiki/SliceTricks/#filtering-without-allocating)

Reply:
Done 👍